### DISCUSSION TITLE
DMS and failover setup

### DISCUSSION THREAD

**Discussion by MarcS1975** [docker-mailserver/docker-mailserver#3742] [Q&A]
How can I setup DMS with failover functionality, takeing advantage of the priority levels in MX records? 
Say I have 2 mailservers MX record prio 10, MX record prio 20, then 20 will only be used for mail delivery when server 10 is down. In that case, server 20 must have all live data replicated from server 10 before it went down. 
How is this achieved in practice? A simple encrypted rsync every minute between servers or is there a more specialised tool? 
Or imapsync?
thanks!

**Comment by polarathene** [comment]
> In that case, server 20 must have all live data replicated from server 10 before it went down.
> How is this achieved in practice? A simple encrypted rsync every minute between servers or is there a more specialised tool?

You probably want this feature: https://github.com/docker-mailserver/docker-mailserver/issues/2048

Add your 👍 and subscribe to that. The author is waiting on me to merge updates for LDAP support before they continue contributing support for replication.

I don't have expertise in this area so have no other input beyond that. I know that if using remote filesystems like NFS there can be some compatibility issues. You also do not want to sync state that belongs only to the container processes at run-time, which is mostly in our `mail-state` volume. That will contain some state like the Postfix mail queue, so that if the container crashed it would not lose mail queued to be delivered/sent, I don't think that'd make sense to sync though.

Accounts would, but when changes are detected to the file-based accounts provisioner (or any monitored files for that matter), the change detection service will process that and it is a bit naive. If that happens too frequently the minor downtime of services may be an issue (_which your failover setup may actually help handle actually_). Presently LDAP usage disables the change detection service, it's planned to be compatible in future.

---

Remote file system concerns aside, if those were handled you could probably use a service that provides a single `/var/mail` for mail storage, where that service offers features like regular backups/redundancy for you. We don't have any official support for remote file systems, and I am aware that there's been some issues with them in the past (_Dovecot docs specifically mention some incompatibilities with NFS or the need to adjust some configuration_).

**Comment by MarcS1975** [comment]
ok - in the interim, I guess, a manual switch over in case of a breakdown is the next best alternative. Is it sufficient to frequently rsync the mapped docker folders **mail-data, config-logs and mail-config** from DMS1 to DMS2, assuming that DMS2 is already running in accordance with the equivalent compose seetings of DMS1?
